projects/dsp-ui/src/lib/viewer/services/event-bus.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Subject, Subscription } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+/**
+ * https://stackoverflow.com/questions/56290722/how-pass-a-event-from-deep-nested-child-to-parent-in-angular-2
+ * This service is used as a way to enable components to communicate with each other no matter how nested they are.
+ * This is intended to provide a cleaner way to emit events from nested components than chaining '@Outputs'.
+ * The EventBus essentially creates a direct communication channel between the emitting component and the listening component.
+ */
+export class EventBusService {
+
+    // Create a subject to hold data which can be subscribed to.
+    // You only get the data after you subscribe.
+    private subject$ = new Subject();
+
+    // Used in the listening component.
+    // i.e. this.eventBusSubscription = this._eventBusService.on(Events.ValueAdded, () => doSomething());
+    on(event: Events, action: any): Subscription {
```

Comment:
done in 3028dae

Reply:
Do you have a use case for `EmitEvent`'s `public value?: () => void`?

Reply:
> Do you have a use case for `EmitEvent`'s `public value?: () => void`?

I just want to ensure the user passes in a function without using `any` and I found this https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#return-types-of-callbacks

Reply:
I understand. I am just wondering if `value` is needed after all.

Reply:
It is needed since we are subscribing here; it's essentially the callback. If we subscribed on the consumer level this would move inside the `.subscribe()`